youtube_dl/extractor/gedidigital.py

```diff
@@ -0,0 +1,180 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..compat import compat_str
+
+
+class GediDigitalBaseIE(InfoExtractor):
```

Comment:
what's the point of creating a base class here?

Reply:
because Gedi is a conglomerate media group that is buying various newspaper/wensites in italy and has various different websites so when I started writing the extractor I though that the various websites would require some sort of dedicated extractor for each different website.
While developing I saw that the situation was much easier and clean than expected and (as you can see in the GediDigitalIE extractor with the TESTS and the VALID_URL regex) only the Base Extractor basically was necessary without dedicated extractor for each website of the Gedi media group.

I kinda left the two classes sepatate because I thought about the possibility of a future aquisition by GediGroup of a website/newspaper that requires a dedicated Extractor, or the possibility for some other extractor to use the GediDigitalBase extractor if necessary.

Reply:
for now, merge with the main extractor and if a need for separation arises in the future, this can be done(only the common code).

Reply:
ok, merged the two classes.

and while I was there I added the possibility to generate http direct mp4 urls that I forgot to commit prior.